### PR TITLE
Fix the existing integration tests and add interrupt sync test case

### PR DESCRIPTION
**Pull request by sgandhi1311**
# Description of change
Fix the existing integration tests and add interrupt sync test case

# Manual QA steps
 - Tested locally on dev-vm. Ran all the test cases.
 - Circle-Ci is passing.
 
# Risks
 - Low Risk
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/test_all_fields.py [hunk 1]
```diff
@@ -79,6 +79,10 @@ def test_run(self):
                 # As we can't generate following field by activecampaign APIs and UI, so removed it form expectation list.
                 if stream == "ecommerce_orders":
                     expected_all_keys = expected_all_keys - {'order_products'}
+                elif stream == "contacts":
+                    expected_all_keys = expected_all_keys - {'email_empty'}
+                elif stream == "users":
+                    expected_all_keys = expected_all_keys - {'password_updated_utc_timestamp', 'cdate', 'udate'}
```
Comment (kethan1122) [inline tests/test_all_fields.py:85]:
```suggestion
                expected_all_keys = expected_all_fields - self.MISSING_FIELDS.get(stream, set())
```
This looks bit dirty. you can define `MISSING_FIELDS` as dict with `key` as `stream name` and `value` as `set of missing fields`

File: tests/base.py [hunk 2]
```diff
@@ -604,3 +605,11 @@ def parse_date(self, date_value):
 
         raise NotImplementedError(
             "Tests do not account for dates of this format: {}".format(date_value))
+    
+    def assertIsDateFormat(self, value, str_format):
```
Comment (kethan1122) [inline tests/base.py:609]: this can be defined as static method. this isn't a blocker, just a suggestion

File: tests/test_bookmark.py [hunk 3]
```diff
@@ -101,20 +97,17 @@ def test_run(self):
...
-                    first_bookmark_value_utc = self.convert_state_to_utc(
-                        first_bookmark_value)
-                    second_bookmark_value_utc = self.convert_state_to_utc(
-                        second_bookmark_value)
 
-                    
-                    simulated_bookmark_value = self.convert_state_to_utc(new_states['bookmarks'][stream])
+                    simulated_bookmark_value = new_states['bookmarks'][stream]
+
+                    self.assertIsInstance(first_bookmark_value, str)
```
Comment (kethan1122) [inline tests/test_bookmark.py:109]: same assertions can be added for `second_bookmark_value` as well

File: tests/test_interrupted_sync.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+                            # during the interrupted sync.
+                            self.assertGreaterEqual(record_time, synced_stream_datetime)
+
+                            # Verify resuming sync replicates all records that were found in the full
+                            # sync (non-interrupted)
+                            self.assertIn(record, first_sync_stream_records,
+                                          msg="Unexpected record replicated in resuming sync.")
+                    else:
+                        # FULL_TABLE stream records should be same
+                        self.assertEqual(interrupted_record_count, full_sync_record_count,
```
Comment (kethan1122) [inline tests/test_interrupted_sync.py:217]: for `full table` we can add an assertion to validate that we don't store the bookmark value
[refer this](https://github.com/stitchdata/tap-tester/blob/master/tap_tester/base_suite_tests/interrupted_sync_test.py#L207-L211)

Comment (kethan1122) [inline tests/test_interrupted_sync.py:217]: Not just the count, we have to make sure that records are also same.
[refer this](https://github.com/stitchdata/tap-tester/blob/master/tap_tester/base_suite_tests/interrupted_sync_test.py#L215-L216)

**Review by kethan1122 — CHANGES_REQUESTED** [review]

**Comment by kethan1122** [comment]
please make sure you have all the assertions mentioned in [base suite](https://github.com/stitchdata/tap-tester/blob/master/tap_tester/base_suite_tests/interrupted_sync_test.py#L122) for interrupted sync

File: tests/test_interrupted_sync.py [hunk 5]
```diff
@@ -0,0 +1,218 @@
+from tap_tester import runner, connections, menagerie, LOGGER
+from base import ActiveCampaignTest
+from dateutil import parser as parser
```
Comment (rdeshmukh15) [inline tests/test_interrupted_sync.py:3]: This import is unused.

Comment (sgandhi1311) [reply tests/test_interrupted_sync.py]: removed

File: tests/test_bookmark.py (hunk 3 above)
Comment (sgandhi1311) [reply tests/test_bookmark.py:109]: done

File: tests/base.py (hunk 2 above)
Comment (sgandhi1311) [reply tests/base.py:610]: done

**Review by kethan1122 — APPROVED** [review]

**Review by rdeshmukh15 — APPROVED** [review]

**Pull request merged**